src/direct-linking/background/storage.test.ts

```diff
@@ -0,0 +1,155 @@
+import indexedDB from 'fake-indexeddb'
+import IDBKeyRange from 'fake-indexeddb/lib/FDBKeyRange'
+import normalize from '../../util/encode-url-for-id'
+
+import Storage from '../../search/search-index-new/storage'
+import { StorageManager } from '../../search/search-index-new/storage/manager'
+// import { AnnotationStorage } from './storage'
+import AnnotationBackround from './'
+
+import * as dummy from './storage.test.data'
+
+const runSuite = () => {
+    const storageManager = new StorageManager()
+    const annotationStorage = new AnnotationBackround({ storageManager })
+        .annotationStorage
+
+    async function insertTestData() {
+        // Insert annotations and direct links
+        annotationStorage.insertDirectLink(dummy.directLink)
+        annotationStorage.createAnnotation(dummy.highlight)
+        annotationStorage.createAnnotation(dummy.annotation)
+        annotationStorage.createAnnotation(dummy.comment)
+        // Insert tags
+        annotationStorage.modifyTags(true)(dummy.tag1, dummy.annotation.url)
+        annotationStorage.modifyTags(true)(dummy.tag2, dummy.annotation.url)
+    }
+
+    async function resetTestData(dbName = 'Memex') {
+        indexedDB.deleteDatabase(dbName)
+
+        // Passing fake IndexedDB to the storage manager
+        storageManager._finishInitialization(
+            new Storage({
+                indexedDB,
+                IDBKeyRange,
+                dbName,
+                storageManager,
+            }),
+        )
+
+        await insertTestData()
+    }
+
+    beforeAll(async () => {
+        await resetTestData()
+    })
+
+    describe('Read operations: ', () => {
```

Comment:
These read ops tests look good! You should probably assert the content of both the fetched annotations and tags as well to have better coverage over these functionality (say I make a change later that unintentionally excludes the annotation's `body` from being returned, for example, but everything else is the same - these tests would still pass right now).